src/main/java/org/terasology/wildAnimalsGenome/AnimalMatingAuthoritySystem.java

```diff
@@ -238,8 +218,6 @@ public void cleanupAfterMating(MatingCleanupEvent event, EntityRef entityRef) {
         EntityRef animal2 = event.animal2;
         animal1.removeComponent(MatingBehaviorComponent.class);
         animal2.removeComponent(MatingBehaviorComponent.class);
-        animal1.send(new UpdateBehaviorEvent());
-        animal2.send(new UpdateBehaviorEvent());
 
         MatingComponent matingComponent1 = animal1.getComponent(MatingComponent.class);
```

Comment:
`MatingBehaviorComponent` removed  from `animal1` and `animal2` before this, hence `matingComponent1` and the further ones would be `NULL`, this results in weird behaviors for both of the parents after the offspring is created. So, remove the `animal1's` `MatingComponent` only after assigning it to `matingComponent1` and similarly for `animal2`.

Reply:
As far as I remember, `MatingBehaviorComponent` is different from `MatingComponent` as in, `MatingComponent` marks that the animal is in fact capable of mating and `MatingBehaviorComponent` indicates that the animal is currently in the mating process and thus requires the `matingCritter` behavior.
After mating is done, the `MatingBehaviorComponent` is removed and `UpdateBehaviorEvent` is called to reset it's behavior to the default (scared or friendly or whatever) and remove the `matingCritter` behavior.